Enable IRQ_STEER on i.MX8MP

LaurentiuM1234 opened the pull request:
Enable the IRQ_STEER interrupt controller on i.MX8MP.

LaurentiuM1234 committed:
nxp: imx8m: adsp: enable the irqstr interrupt controller

Enable the irqstr interrupt controller for the adsp-based
imx8mp.

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>
Tested-by: Daniel Baluta <daniel.baluta@nxp.com>

dbaluta commented:
@LaurentiuM1234 I will reopen this.  Tested with iMX8MP and SDMA changes and it works fine.

LaurentiuM1234 committed:
drivers: intc: irqstr: initialize 'enabled' variable

Initialize the 'enabled' variable before using it.
This fixes the following compilation warning:

"warning: 'enabled' may be used uninitialized [-Wmaybe-uninitialized]"

issued when compiling with `CONFIG_DEBUG` enabled.

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>
Tested-by: Daniel Baluta <daniel.baluta@nxp.com>

dbaluta commented:
Changes since previous version:

- rebased on top of main
- tested with SOF and openamp on i.MX8MP.
